cmake-modules/AzureVcpkg.cmake

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+# Defines utilities for Azure SDK to integrate with VCPKG automatically
+
+macro(az_vcpkg_integrate)
+  # AUTO CMAKE_TOOLCHAIN_FILE:
+  #   User can call `cmake -DCMAKE_TOOLCHAIN_FILE="path_to_the_toolchain"` as the most specific scenario.
+  #   An env var VCPKG_ROOT or VCPKG_INSTALLATION_ROOT can be set to let Azure SDK to set the VCPKG toolchain automatically.
+  #   As the last alternative (default case), Azure SDK will automatically clone VCPKG folder and set toolchain from there.
+  if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+    if(DEFINED ENV{VCPKG_ROOT})
+      set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
+          CACHE STRING "")
+    elseif(DEFINED ENV{VCPKG_INSTALLATION_ROOT})
+      set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake"
+          CACHE STRING "")
+    else()
+      # Set AZURE_SDK_DISABLE_AUTO_VCPKG env var to avoid Azure SDK from cloning and setting VCPKG automatically
+      # This option delegate package's dependencies installation to user.
+      if(NOT DEFINED ENV{AZURE_SDK_DISABLE_AUTO_VCPKG})
+        # GET VCPKG FROM SOURCE
+        #  User can set env var AZURE_SDK_VCPKG_COMMIT to pick the VCPKG commit to fetch
+        set(VCPKG_COMMIT_STRING f0aa678b7471497f1adedcc99f40e1599ad22f69) # default SDK tested commit
```

Comment:
Rather than picking an arbitrary commit, it would be better to pick one of the released tags.

This one is from April 18:
https://github.com/microsoft/vcpkg/commit/f0aa678b7471497f1adedcc99f40e1599ad22f69

Maybe we pick one of these:
https://github.com/microsoft/vcpkg/tags

The latest one is this:
https://github.com/microsoft/vcpkg/releases/tag/2022.07.25
Commit:
https://github.com/microsoft/vcpkg/commit/f93ba152d55e1d243160e690bc302ffe8638358e

Reply:
I would try to use the same as C++, as that would give cache binaries for C and C++ at the same time for CI =)

Let's start using the same as C++ and the we can update both to something else